IMAGE-CLASSIFICATION/ResNet-PyTorch/main.py

```diff
@@ -0,0 +1,313 @@
+        # Run the training batches
+        t = tqdm(trainloader, desc='epoch:{} loss:{:.4f} accuracy:{}'.format(epoch, 0.0, 'NA'), leave=True)
+        for b, (X_train, y_train) in enumerate(t):
+        
+            # Apply the model
+            X_train = X_train.to(device)
+            y_train = y_train.to(device)
+            y_pred = model(X_train)
+            loss = criterion(y_pred, y_train)
+            running_loss+=loss.item()
+ 
+        # Tally the number of correct predictions
+            predicted = torch.max(y_pred.data, 1)[1]
+            batch_corr = (predicted == y_train).sum()
+            trn_corr += batch_corr.item()
+        
+            # Update parameters
+            optimizer.zero_grad()
+            loss.backward()
+            optimizer.step()
+            scheduler.step()
+        
+        
+            # # Print interim results
+            if b%50 == 0:
+                t.set_description('epoch:{} loss:{:.4f} accuracy:{}'.format(epoch, loss.item(), batch_corr.item()))
+                t.refresh()
+        #     print(f'epoch: {i:2}  batch: {b:4} [{128*b:6}/50000]  loss: {loss.item():10.8f}   accuracy: {(batch_corr.item()*100/128):10.8f}%',end=' ')
+
+        
+        np.append(train_losses,loss.item())
+        np.append(train_correct,trn_corr/500)
+    
+        
+    # Run the testing batches
+        with torch.no_grad():
+            t = tqdm(testloader, desc="[Validation] Epoch:{}".format(epoch), leave=True)
+            for (X_test, y_test) in t:
+          
+                X_test = X_test.to(device)
+                y_test = y_test.to(device)
+                # Apply the model
+                y_val = model(X_test)
+
+                # Tally the number of correct predictions
+                predicted = torch.max(y_val.data, 1)[1] 
+                batch_corr = (predicted == y_test).sum()  
+                tst_corr+=batch_corr.item()
+                # if b==40 :
+                #   print(f'test accuracy :{batch_corr.item()*100/128}% ')
+ 
+        loss = criterion(y_val, y_test)
+        print('Epoch:{}  loss:{:.3f} accuracy:{:.2f}% test_accuracy:{:.2f}%'.format(epoch, running_loss, trn_corr/500, tst_corr/100))
+
+        np.append(test_losses,loss.item())
+        np.append(test_correct,tst_corr/100)    
+    print(f'\nDuration: {(time.time() - start_time)/60} minutes') # print the time elapsed  ,minutes') # print the time elapsed
+    
+    return train_losses,test_losses,train_correct,test_correct
+device = config['run_config']['device']
+
+model.to(device)
+criterion = nn.CrossEntropyLoss()
+train_loader,test_loader = get_loader(config['optim_config']['batch_size'],config['run_config']['num_workers'])
+
+train_loss,test_loss,train_accuracy,test_accuracy = train(model,config['optim_config']['epochs'],train_loader,test_loader,device,criterion,optimizer,scheduler)
+
+plot_loss_epoch(train_loss,test_loss)
+plot_accuracy_epoch(train_accuracy,test_accuracy)
+make_heat_map(model,test_loader,device)
```

Comment:
instead of plotting, save the plot as file.

Reply:
did both plotting and saving plot to the file.
add models.py
add reference to paper and how to run code on README.md